src/microphone/browser_audio_processor.ts

```diff
@@ -0,0 +1,127 @@
+import { ErrDeviceNotSupported, ErrNoAudioConsent, ErrNotInitialized } from './types'
+
+export type AudioHandler = (audioBuffer: Float32Array) => void
+
+export interface AudioProcessor {
+  readonly sampleRate: number
+  initialize(): Promise<void>
+  close(): void
+  mute(): void
+  unmute(): void
+  isMuted(): boolean
+}
+
+const audioProcessEvent = 'audioprocess'
+
+export class BrowserAudioProcessor implements AudioProcessor {
+  private initialized: boolean = false
+  readonly sampleRate: number
+
+  // Audio callback invoked when audio is returned by the script processor.
+  private readonly isSafari: boolean
```

Comment:
Low quality nitpick: this could be `isWebkit`

Reply:
Sounds good, I'll update that 👍

Reply:
Updated!

Reply:
Updated!